### DISCUSSION TITLE
Incorporating a new "stl" image for test case 8 (Dissolution of calcite post with ALE)

### DISCUSSION THREAD

**Discussion by ritik-kgp** [GeoChemFoam/GeoChemFoam#53] [Q&A]
I am required to show how the solution chemistry modifies, along with dissolution of basalt rock when acidic water flows through the **fractures** of basalt rock .  For this purpose, I considered test case 8 as the most suitable starting point.  I am beginning with changing the image (with fractures to best suit the requirements), I need guidance regarding how to modify the image: 
1) Here raw image is being used, can I use my own "stl" image? In this case, which files I should consider for changes? And any other precautions to take care of?
2) Otherwise, shoukd I duplicate the existing calcite rock image at the corners of the blockMesh, so as to obtain the geometry similar as fractures? 

Your response is highly appreciated. Thanks!!

**Comment by GeoChemFoam** [comment]
Hi,

You can easily change to stl where you specify the format. Did you read the script all the way through?

There are a few things you will need to change in create mesh to make it work. However, this is a research code and we do not provide support for this.

Best wishes,

Hannah

From: Ritik Chaturvedi ***@***.***>
Date: Sunday, 10 November 2024 at 05:04
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GeoChemFoam/GeoChemFoam] Incorporating a new "stl" image for test case 8 (Dissolution of calcite post with ALE) (Discussion #53)
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


I am required to show how the solution chemistry modifies, along with dissolution of basalt rock when acidic water flows through the fractures of basalt rock . For this purpose, I considered test case 8 as the most suitable starting point. I am beginning with changing the image (with fractures to best suit the requirements), I need guidance regarding how to modify the image:

  1.  Here raw image is being used, can I use my own "stl" image? In this case, which files I should consider for changes? And any other precautions to take care of?
  2.  Otherwise, shoukd I duplicate the existing calcite rock image at the corners of the blockMesh, so as to obtain the geometry similar as fractures?

Your response is highly appreciated. Thanks!!

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/53>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFRKRT5KLKNGDW6VMHDZ73SPJAVCNFSM6AAAAABRP2N5PGVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGQ2TGMBUGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by ritik-kgp** [comment]
Thanks for your response. So am I not allowed to use GeoChemFoam?

**Comment by GeoChemFoam** [comment]
You are welcome to use it.

We just don’t have time to provide support for applications outside the scope of the tutorials.

Best wishes,

Hannah

From: Ritik Chaturvedi ***@***.***>
Date: Monday, 11 November 2024 at 11:06
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Menke, Hannah ***@***.***>, Comment ***@***.***>
Subject: Re: [GeoChemFoam/GeoChemFoam] Incorporating a new "stl" image for test case 8 (Dissolution of calcite post with ALE) (Discussion #53)
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Thanks for your response. So am I not allowed to use GeoChemFoam?

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/53#discussioncomment-11213417>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFTSDIAP7FGDD4PC7TT2ACFSRAVCNFSM6AAAAABRP2N5PGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRRGM2DCNY>.
You are receiving this because you commented.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by GeoChemFoam** [comment]
Hi,

Further to my previous email, you can see in createmesh that you can specify stl as an input format.



Best wishes,

Hannah

From: Menke, Hannah ***@***.***>
Date: Monday, 11 November 2024 at 11:11
To: GeoChemFoam/GeoChemFoam ***@***.***>, GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Comment ***@***.***>
Subject: Re: [GeoChemFoam/GeoChemFoam] Incorporating a new "stl" image for test case 8 (Dissolution of calcite post with ALE) (Discussion #53)
You are welcome to use it.

We just don’t have time to provide support for applications outside the scope of the tutorials.

Best wishes,

Hannah

From: Ritik Chaturvedi ***@***.***>
Date: Monday, 11 November 2024 at 11:06
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Menke, Hannah ***@***.***>, Comment ***@***.***>
Subject: Re: [GeoChemFoam/GeoChemFoam] Incorporating a new "stl" image for test case 8 (Dissolution of calcite post with ALE) (Discussion #53)
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Thanks for your response. So am I not allowed to use GeoChemFoam?

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/53#discussioncomment-11213417>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFTSDIAP7FGDD4PC7TT2ACFSRAVCNFSM6AAAAABRP2N5PGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRRGM2DCNY>.
You are receiving this because you commented.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by ritik-kgp** [comment]
Thank you for your previous response. I just duplicated the calcite post in "Dissolution of calcite post with ALE" tutorial. The mesh look like as shown in image attached. 
![Screenshot from 2024-11-22 16-11-59](https://github.com/user-attachments/assets/93196ecb-9a69-4ff8-9fe8-b135bb411845)
Here, the issue is **there occurs no dissolution** even if I run the simulation for 20000 seconds. The calcite posts remain same throughout, unlike the case with original tutorial where whole dissolution happens in around 10000 seconds. There occurs no error even. I hope you can suggest me where it is happening wrong. It shall be very helpful. Thank you and Regards.
BTW, I just made changes in createMesh.sh(translated stl at two different positions) and snappyHexMeshDict1 to obtained this mesh(as shown in image).

**Comment by GeoChemFoam** [comment]
What else did you change other than the image? The resolution? The flowrate? The reaction rate?

It is impossible to tell what is wrong without this information.

Try changing only one parameter at a time and work step by step towards your goal.


From: Ritik Chaturvedi ***@***.***>
Date: Friday, 22 November 2024 at 10:57
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Menke, Hannah ***@***.***>, Comment ***@***.***>
Subject: Re: [GeoChemFoam/GeoChemFoam] Incorporating a new "stl" image for test case 8 (Dissolution of calcite post with ALE) (Discussion #53)
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Thank you for your previous response. I just duplicated the calcite post in "Dissolution of calcite post with ALE" tutorial. The mesh look like as shown in image attached.
Screenshot.from.2024-11-22.16-11-59.png (view on web)<https://github.com/user-attachments/assets/93196ecb-9a69-4ff8-9fe8-b135bb411845>
Here, the issue is there occurs no dissolution even if I run the simulation for 20000 seconds. The calcite posts remain same throughout, unlike the case with original tutorial where whole dissolution happens in around 10000 seconds. There occurs no error even. I hope you can suggest me where it is happening wrong. It shall be very helpful. Thank you and Regards.
BTW, I just made changes in createMesh.sh(translated stl at two different positions) and snappyHexMeshDict1 to obtained this mesh(as shown in image).

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/53#discussioncomment-11347740>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFU6N37LHJZ37K443QL2B4E2HAVCNFSM6AAAAABRP2N5PGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZUG43TIMA>.
You are receiving this because you commented.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by ritik-kgp** [comment]
Thank you for responding.  As far as changes are considered, I made no changes in resolution, flowrate and reaction rate. I only made changes in createMesh.sh and snappyHexMeshDict1 file inside system folder.   The **changed createMesh.sh** file is mentioned below, with the changes made bold and italic for ease to find. 
#!/bin/bash

###### USERS INPUT ############################################################

#Define image name
Image_name="calcitePost"

#Image directory location ($PWD if current directory)
dir="$GCFOAM_IMG/raw"

#Choose image format
format='raw'

#Choose if the image is compressed or not
compressed='yes'

#If format='raw', define image dimensions
x_dim=536
y_dim=300
z_dim=40

#If format='raw', define value of the pores and solid in the image
pores_value=255
solid_value=0

#if format= 'stl' enter the location of a pore
#pore_index_X=0.0001522 #0
#pore_index_Y=0.0001522 #0
#pore_index_Z=0

#define resolution (m)
res=0.000005

#Define cropping parameters
x_min=0
x_max=536
y_min=0
y_max=300
z_min=0
z_max=40

#number of cells of initial mesh
n_x=134
n_y=75
n_z=10

#Level of refinement - mesh is refined by n_level at the pore surfaces
n_level=1

#flow direction (0 is x, 1 is y and 2 is z)
direction=0

#Number of processors
NP=8

#### END OF USER INPUT #######################################################

_**translations=(
	"(-120 0 -0.5)"
	"(120 0 -0.5)"
)**_

if [ $format != 'raw' ] && [ $format != 'stl' ]
then
        echo "ERROR: only raw and stl format are implemented for this solver"
        exit
fi

filename=$Image_name\.$format

if [ $compressed == 'yes' ]
then
        filename=$Image_name\.$format\.tar.gz
fi

cp $dir\/$filename constant/triSurface/.
cd constant/triSurface
if [ $compressed == 'yes' ]
then
        tar -xf $filename
fi

if [ $format == 'raw' ]
then
    echo -e "make stl"
    python raw2stl.py --x_min=$x_min --x_max=$x_max --y_min=$y_min --y_max=$y_max --z_min=$z_min --z_max=$z_max --pores_value=$pores_value --solid_value=$solid_value  --image_name=$Image_name --x_dim=$x_dim --y_dim=$y_dim --z_dim=$z_dim
    rm $Image_name\.*

**_//commented below lines and added for loop at the end to account for both images_**
 **#surfaceTransformPoints -translate '(-0.5 -0.5 -0.5)' Image_meshed.stl Image_meshed.stl > ../../surfaceTransformPoints1.out
 #vector="($res $res $res)"
 #surfaceTransformPoints -scale "$vector" Image_meshed.stl Image_meshed.stl > ../../surfaceTransformPoints2.out**
    pore_index_X="$(cat pore_indx)"
    pore_index_Y="$(cat pore_indy)"
    pore_index_Z="$(cat pore_indz)"
    pore_index_0=$(expr $pore_index_X*$res | bc)
    pore_index_1=$(expr $pore_index_Y*$res | bc)
    pore_index_2=$(expr $pore_index_Z*$res | bc)


elif [ $format == 'stl' ]
then
    echo -e "reading stl"
    mv $Image_name\.stl Image_meshed.stl
    rm -f $Image_name\.*
    tranX=$(expr -1*$x_min | bc)
    tranY=$(expr -1*$y_min | bc)
    tranZ=$(expr -1*$z_min | bc)
    vector="($tranX $tranY $tranZ)"
    surfaceTransformPoints -translate "$vector" Image_meshed.stl Image_meshed.stl > ../../surfaceTransformPoints1.out
    vector="($res $res $res)"
    surfaceTransformPoints -scale "$vector" Image_meshed.stl Image_meshed.stl > ../../surfaceTransformPoints2.out
    pore_index_0=$(expr $res*$pore_index_X-$res*$x_min | bc)
    pore_index_1=$(expr $res*$pore_index_Y-$res*$y_min | bc)
    pore_index_2=$(expr $res*$pore_index_Z-$res*$z_min | bc)
fi

echo -e "Coordinates at center of a pore = ($pore_index_0,$pore_index_1,$pore_index_2) in the cropped image"

**_//added for loop_**
**_//translating and scaling each calcite post
for i in "${!translations[@]}";do
	echo "Translating calcite post $((i+1))"
	surfaceTransformPoints -translate "${translations[$i]}" Image_meshed.stl "Image_meshed${i}.stl" > "../../surfaceTransformPoints_$((i+1)).out"
	vector="($res $res $res)"
	surfaceTransformPoints -scale "$vector" "Image_meshed${i}.stl" "Image_meshed${i}.stl" > "../../surfaceTransformPoints_$((i+1))_scaled.out"
done_**

cd ../..

 
#Create background mesh
echo -e "Create background mesh"
cp system/blockMeshDict$direction system/blockMeshDict

dx=$(expr $x_max-1*$x_min | bc)
dy=$(expr $y_max-1*$y_min | bc)
dz=$(expr $z_max-1*$z_min | bc)

sed -i "s/dx/$dx/g" system/blockMeshDict
sed -i "s/dy/$dy/g" system/blockMeshDict
sed -i "s/dz/$dz/g" system/blockMeshDict

sed -i "s/nx/$n_x/g" system/blockMeshDict
sed -i "s/ny/$n_y/g" system/blockMeshDict
sed -i "s/nz/$n_z/g" system/blockMeshDict

sed -i "s/res/$res/g" system/blockMeshDict

cp system/controlDict0 system/controlDict

blockMesh  > blockMesh.out

cp system/snappyHexMeshDict1 system/snappyHexMeshDict

sed -i "s/nlevel/$n_level/g" system/snappyHexMeshDict

sed -i "s/poreIndex0/$pore_index_0/g" system/snappyHexMeshDict
sed -i "s/poreIndex1/$pore_index_1/g" system/snappyHexMeshDict
sed -i "s/poreIndex2/$pore_index_2/g" system/snappyHexMeshDict

x_1=0
y_1=0
z_1=0

xSize=$(echo "$x_max - $x_min" | bc)
ySize=$(echo "$y_max - $y_min" | bc)
zSize=$(echo "$z_max - $z_min" | bc)

x_2=$(expr $xSize*$res | bc)
y_2=$(expr $ySize*$res | bc)
z_2=$(expr $zSize*$res | bc)


cp system/postProcessDict1 system/postProcessDict
sed -i "s/x_1/$x_1/g" system/postProcessDict
sed -i "s/y_1/$y_1/g" system/postProcessDict
sed -i "s/z_1/$z_1/g" system/postProcessDict

sed -i "s/x_2/$x_2/g" system/postProcessDict
sed -i "s/y_2/$y_2/g" system/postProcessDict
sed -i "s/z_2/$z_2/g" system/postProcessDict

sed -i "s/flowdir/$direction/g" system/postProcessDict

if [ $NP -gt 1 ]
then
    cp system/decomposeParDict1 system/decomposeParDict
    sed -i "s/NP/$NP/g" system/decomposeParDict
    #Decompose background mesh
    echo -e "Decompose background mesh"
    decomposePar > decomposeBlockMesh.out
    rm -rf constant/polyMesh
fi

echo -e "BlockMesh and Image created. Ready for snappyHeMesh" 


The changed portion of snappyHexMeshDict1 file is given below:

geometry
{

    Image_meshed0.stl
    {
        type triSurfaceMesh;
        name movingWalls;
    }

    Image_meshed1.stl
    {
	    type triSurfaceMesh;
	    name movingWalls;
    }
};

**Comment by GeoChemFoam** [comment]
You still have image format set to raw? and compressed to yes? You also have not given the xcoordinates at the center of a pore

Best wishes,

Hannah

From: Ritik Chaturvedi ***@***.***>
Date: Friday, 22 November 2024 at 16:06
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Menke, Hannah ***@***.***>, Comment ***@***.***>
Subject: Re: [GeoChemFoam/GeoChemFoam] Incorporating a new "stl" image for test case 8 (Dissolution of calcite post with ALE) (Discussion #53)
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Thank you for responding. As far as changes are considered, I made no changes in resolution, flowrate and reaction rate. I only made changes in createMesh.sh and snappyHexMeshDict1 file inside system folder. The changed createMesh.sh file is mentioned below, with the changes made bold and italic for ease to find.
#!/bin/bash

USERS INPUT

#Define image name
Image_name="calcitePost"

#Image directory location ($PWD if current directory)
dir="$GCFOAM_IMG/raw"

#Choose image format
format='raw'

#Choose if the image is compressed or not
compressed='yes'

#If format='raw', define image dimensions
x_dim=536
y_dim=300
z_dim=40

#If format='raw', define value of the pores and solid in the image
pores_value=255
solid_value=0

#if format= 'stl' enter the location of a pore
#pore_index_X=0.0001522 #0
#pore_index_Y=0.0001522 #0
#pore_index_Z=0

#define resolution (m)
res=0.000005

#Define cropping parameters
x_min=0
x_max=536
y_min=0
y_max=300
z_min=0
z_max=40

#number of cells of initial mesh
n_x=134
n_y=75
n_z=10

#Level of refinement - mesh is refined by n_level at the pore surfaces
n_level=1

#flow direction (0 is x, 1 is y and 2 is z)
direction=0

#Number of processors
NP=8

END OF USER INPUT

translations=(
"(-120 0 -0.5)"
"(120 0 -0.5)"
)

if [ $format != 'raw' ] && [ $format != 'stl' ]
then
echo "ERROR: only raw and stl format are implemented for this solver"
exit
fi

filename=$Image_name.$format

if [ $compressed == 'yes' ]
then
filename=$Image_name.$format.tar.gz
fi

cp $dir/$filename constant/triSurface/.
cd constant/triSurface
if [ $compressed == 'yes' ]
then
tar -xf $filename
fi

if [ $format == 'raw' ]
then
echo -e "make stl"
python raw2stl.py --x_min=$x_min --x_max=$x_max --y_min=$y_min --y_max=$y_max --z_min=$z_min --z_max=$z_max --pores_value=$pores_value --solid_value=$solid_value --image_name=$Image_name --x_dim=$x_dim --y_dim=$y_dim --z_dim=$z_dim
rm $Image_name.*

//commented below lines and added for loop at the end to account for both images
#surfaceTransformPoints -translate '(-0.5 -0.5 -0.5)' Image_meshed.stl Image_meshed.stl > ../../surfaceTransformPoints1.out
#vector="($res $res $res)"
#surfaceTransformPoints -scale "$vector" Image_meshed.stl Image_meshed.stl > ../../surfaceTransformPoints2.out
pore_index_X="$(cat pore_indx)"
pore_index_Y="$(cat pore_indy)"
pore_index_Z="$(cat pore_indz)"
pore_index_0=$(expr $pore_index_X*$res | bc)
pore_index_1=$(expr $pore_index_Y*$res | bc)
pore_index_2=$(expr $pore_index_Z*$res | bc)

elif [ $format == 'stl' ]
then
echo -e "reading stl"
mv $Image_name.stl Image_meshed.stl
rm -f $Image_name.*
tranX=$(expr -1*$x_min | bc)
tranY=$(expr -1*$y_min | bc)
tranZ=$(expr -1*$z_min | bc)
vector="($tranX $tranY $tranZ)"
surfaceTransformPoints -translate "$vector" Image_meshed.stl Image_meshed.stl > ../../surfaceTransformPoints1.out
vector="($res $res $res)"
surfaceTransformPoints -scale "$vector" Image_meshed.stl Image_meshed.stl > ../../surfaceTransformPoints2.out
pore_index_0=$(expr $res*$pore_index_X-$res*$x_min | bc)
pore_index_1=$(expr $res*$pore_index_Y-$res*$y_min | bc)
pore_index_2=$(expr $res*$pore_index_Z-$res*$z_min | bc)
fi

echo -e "Coordinates at center of a pore = ($pore_index_0,$pore_index_1,$pore_index_2) in the cropped image"

//added for loop
//translating and scaling each calcite post
for i in "${!translations[@]}";do
echo "Translating calcite post $((i+1))"
surfaceTransformPoints -translate "${translations[$i]}" Image_meshed.stl "Image_meshed${i}.stl" > "../../surfaceTransformPoints$((i+1)).out"
vector="($res $res $res)"
surfaceTransformPoints -scale "$vector" "Image_meshed${i}.stl" "Image_meshed${i}.stl" > "../../surfaceTransformPoints_$((i+1))scaled.out"
done

cd ../..

#Create background mesh
echo -e "Create background mesh"
cp system/blockMeshDict$direction system/blockMeshDict

dx=$(expr $x_max-1*$x_min | bc)
dy=$(expr $y_max-1*$y_min | bc)
dz=$(expr $z_max-1*$z_min | bc)

sed -i "s/dx/$dx/g" system/blockMeshDict
sed -i "s/dy/$dy/g" system/blockMeshDict
sed -i "s/dz/$dz/g" system/blockMeshDict

sed -i "s/nx/$n_x/g" system/blockMeshDict
sed -i "s/ny/$n_y/g" system/blockMeshDict
sed -i "s/nz/$n_z/g" system/blockMeshDict

sed -i "s/res/$res/g" system/blockMeshDict

cp system/controlDict0 system/controlDict

blockMesh > blockMesh.out

cp system/snappyHexMeshDict1 system/snappyHexMeshDict

sed -i "s/nlevel/$n_level/g" system/snappyHexMeshDict

sed -i "s/poreIndex0/$pore_index_0/g" system/snappyHexMeshDict
sed -i "s/poreIndex1/$pore_index_1/g" system/snappyHexMeshDict
sed -i "s/poreIndex2/$pore_index_2/g" system/snappyHexMeshDict

x_1=0
y_1=0
z_1=0

xSize=$(echo "$x_max - $x_min" | bc)
ySize=$(echo "$y_max - $y_min" | bc)
zSize=$(echo "$z_max - $z_min" | bc)

x_2=$(expr $xSize*$res | bc)
y_2=$(expr $ySize*$res | bc)
z_2=$(expr $zSize*$res | bc)

cp system/postProcessDict1 system/postProcessDict
sed -i "s/x_1/$x_1/g" system/postProcessDict
sed -i "s/y_1/$y_1/g" system/postProcessDict
sed -i "s/z_1/$z_1/g" system/postProcessDict

sed -i "s/x_2/$x_2/g" system/postProcessDict
sed -i "s/y_2/$y_2/g" system/postProcessDict
sed -i "s/z_2/$z_2/g" system/postProcessDict

sed -i "s/flowdir/$direction/g" system/postProcessDict

if [ $NP -gt 1 ]
then
cp system/decomposeParDict1 system/decomposeParDict
sed -i "s/NP/$NP/g" system/decomposeParDict
#Decompose background mesh
echo -e "Decompose background mesh"
decomposePar > decomposeBlockMesh.out
rm -rf constant/polyMesh
fi

echo -e "BlockMesh and Image created. Ready for snappyHeMesh"

The changed portion of snappyHexMeshDict1 file is given below:

geometry
{

Image_meshed0.stl

{

    type triSurfaceMesh;

    name movingWalls;

}



Image_meshed1.stl

{

    type triSurfaceMesh;

    name movingWalls;

}

};

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/53#discussioncomment-11350807>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFTHZCOG2273KIDOLF32B5JBZAVCNFSM6AAAAABRP2N5PGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZVGA4DANY>.
You are receiving this because you commented.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Comment by GeoChemFoam** [comment]
An stl is just a surface. The model has no way of knowing which side is pore and which side is grain. By providing a coordinate at the center of the pore the model knows which side to mesh.

Best wishes,

Hannah

From: Ritik Chaturvedi ***@***.***>
Date: Saturday, 23 November 2024 at 05:14
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Menke, Hannah ***@***.***>, Comment ***@***.***>
Subject: Re: [GeoChemFoam/GeoChemFoam] Incorporating a new "stl" image for test case 8 (Dissolution of calcite post with ALE) (Discussion #53)
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


I still have image format set to raw, I am using raw2stl.py file to convert it to stl, then translating this stl to two positions inside the block. I did not understood your point regarding "x-coordinate definition of the pore", can you please elaborate a little if possible? Thanks

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/53#discussioncomment-11354188>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFRVKZXIYLJMHMWECIL2B66WTAVCNFSM6AAAAABRP2N5PGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZVGQYTQOA>.
You are receiving this because you commented.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.
